src/SelfServe/SelfServeUtils.tsx

```diff
@@ -0,0 +1,210 @@
+import "reflect-metadata";
+import {
+  ChoiceItem,
+  Node,
+  Info,
+  InputTypeValue,
+  Descriptor,
+  AnyInput,
+  NumberInput,
+  StringInput,
+  BooleanInput,
+  ChoiceInput,
+  InputType
+} from "../Explorer/Controls/SmartUi/SmartUiComponent";
+
+const SelfServeType = "selfServeType";
+
+export class SelfServeBase {
+  public static toSmartUiDescriptor(): Descriptor {
+    return Reflect.getMetadata(this.name, this) as Descriptor;
+  }
+}
+
+export interface CommonInputTypes {
+  id: string;
```

Comment:
Now All decorators are coalesced into one - @Values. this will allow for type checking and ensure the right set of values are used. WIll add @Values in next iteration